### PR TITLE
[SPARK-725] Add quota setup to deploy-dispatchers script

### PR DESCRIPTION
**Pull request by sschneid**
https://jira.mesosphere.com/browse/SPARK-725

### CODE REVIEW

File: scale-tests/deploy-dispatchers.py [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+    --kerberos-realm <realm>     the Kerberos realm used to render the principal
+    --log-level <level>          log level [default: INFO]
+    --mem <n>                    amount of memory (mb) to use per dispatcher [default: 1024.0]
+    --options-json <file>        a file containing installation options in JSON format
+    --package-name <name>        name of the Spark package name [default: spark]
+    --package-repo <url>         URL of the Spark package repo to install from
+    --quota-dispatcher-cpus <n>  number of CPUs to use for dispatcher quota [default: 1]
+    --quota-dispatcher-mem <n>   amount of memory (mb) to use per dispatcher quota [default: 2048.0]
+    --quota-driver-cpus <n>      number of CPUs to use for driver quota [default: 1]
+    --quota-driver-mem <n>       amount of memory (mb) to use per driver quota [default: 1024.0]
```
Comment (susanxhuynh) [inline scale-tests/deploy-dispatchers.py]: Could you change the default to 1524.0? I realized after testing that there's some memory overhead I hadn't accounted for.

File: scale-tests/deploy-dispatchers.py [hunk 2]
```diff
@@ -1,3 +1,46 @@
...
+    --history-service <url>      URL of the Spark history service
+    --kdc-hostname <hostname>    the name or address of a host running a KDC
+    --kdc-port <port>            the port of the host running a KDC [default: 88]
+    --kerberos-realm <realm>     the Kerberos realm used to render the principal
+    --log-level <level>          log level [default: INFO]
+    --mem <n>                    amount of memory (mb) to use per dispatcher [default: 1024.0]
+    --options-json <file>        a file containing installation options in JSON format
+    --package-name <name>        name of the Spark package name [default: spark]
+    --package-repo <url>         URL of the Spark package repo to install from
+    --quota-dispatcher-cpus <n>  number of CPUs to use for dispatcher quota [default: 1]
```
Comment (susanxhuynh) [inline scale-tests/deploy-dispatchers.py]: It wasn't in my example, but we'll want quota for GPUs soon, with a default of 0.

**Comment by susanxhuynh** [comment]
@sschneid One more note: when I was running this manually, I had to save the driver roles for use in the "scale_test.py" script. 
Also, did you want to remove the roles in the uninstall-dispatchers.sh script?

**Comment by sschneid** [comment]
Thanks @susanxhuynh!  I've adjusted defaults, added support for GPU quotas, and am writing the dispatcher role in the outfile.

I'd like to update `uninstall-dispatchers.sh` to be a Python script (like the others in `scale-tests/`) in a forthcoming PR, any objections?

**Comment by susanxhuynh** [comment]
@sschneid  No objection to making it a Python script.
What do you want to do about consuming the driver role in the test script? Separate PR?

**Comment by sschneid** [comment]
> What do you want to do about consuming the driver role in the test script? Separate PR?

I've added a commit to update `scale_test.py`.

**Pull request merged**